R/logit_utils.R

```diff
@@ -0,0 +1,83 @@
+#' @title Logit and inverse logit functions
+#'
+#' @description Logit and inverse logit functions, with options to scale input
+#'   to logit, scale output of inverse logit, or perform a transformation
+#'   such that the domain in non-logit space is as specified (standard
+#'   is (0, 1)).
+#'
+#' @param x \[`numeric()`\]\cr Value to take the logit or inverse logit of.
+#' @param scale \[`numeric()`\]\cr Value to scale by. Default 1.
+#' @param domain_lower \[`numeric()`\]\cr Lower bound of domain in non-logit
+#'   space, inclusive. Default 0. Logit at lower bound is -Inf.
+#' @param domain_upper \[`numeric()`\]\cr Upper bound of domain in non-logit
+#'   space, inclusive. Default 1. Logit at upper bound is Inf.
+#'
+#' @return \[`numeric()`\]\cr The calculated logit or inverse logit value.
+#'
+#' @details
+#' Standard:
+#'
+#' * logit (x) = log (x / (1-x))
+#' * inverse logit (x) = exp (x) / (1 + exp (x))
+#'
+#' Optional scaled logit with 'scale' argument:
+#'
+#' * logit (x, s) = log (s * x / (1 - s * x))
+#' * inverse logit (x, s) = (1 / s) * exp (x) / (1 + exp (x))
+#'
+#' Optional logit with transformed domain:
+#'
+#' * logit (x, l, u) = log(x' / (1 - x')) where x' = (x - l) / (u - l)
+#' * inverse logit (x, l, u) = [ exp (x) / (1 + exp (x)) ] * (u - l) + l
+#'
+#' Other notes: Scaled values of x outside of (`domain_lower`, `domain_upper`)
+#'   will return NaN and result in a warning from `logit` function.
+#'
+#' @examples
+#' # Standard
+#' logit(0.1)
+#' invlogit(-2)
+#'
+#' # Scaled
+#' x <- c(10, 20, 30)
+#' logit_x <- logit(x, scale = 0.01)
```

Comment:
So this is equivalent to `logit(x, domain_upper = 100)`. When would we actually want to use the `scale` argument, for percentages? I wonder if its clearer to just have `domain_lower` and `domain_upper` arguments?

Also does there need to be a check whether someone tries to use all these arguments?

Reply:
This is why I was asking folks about how they use scaled logits. As far as I can tell we mostly use the domain change, but I don't mind keeping `scale` in there as an option to make these functions more flexible.

Reply:
Also: you could technically specify both `scale` and `domain_lower` / `domain_upper`.

Reply:
I guess I was just thinking that it took me a while to figure out what was going on with the two options so might be clearer to just have the one option that seems to be flexible for both use cases. 

`scale` isn't really doing much anyway where someone could do that before hand if it was really needed.